src/herder/HerderSCPDriver.cpp

```diff
@@ -976,11 +1018,24 @@ HerderSCPDriver::recordSCPExternalizeEvent(uint64_t slotIndex, NodeID const& id,
                 make_optional<VirtualClock::time_point>(now);
         }
     }
-    else if (timing.mSelfExternalize)
+    else
     {
-        recordLogTiming(*timing.mSelfExternalize, now,
-                        mSCPMetrics.mExternalizeDelay,
-                        fmt::format("externalize delay ({})",
+        // Record externalize delay
+        if (timing.mSelfExternalize)
+        {
+            recordLogTiming(*timing.mSelfExternalize, now,
+                            mSCPMetrics.mExternalizeDelay,
+                            fmt::format("externalize delay ({})",
+                                        mApp.getConfig().toShortString(id)),
+                            std::chrono::nanoseconds::zero(), slotIndex);
+        }
+
+        // Record lag for other nodes
+        auto recordedLag =
```

Comment:
why not just use `medida::Timer` directly and avoid `shared_ptr`, then you can just do
`auto &vLag = mQSetLag[id];`